File: Coding Sample/MS.NET/Fundamentals/Generic/Basics/genmethodtest2.cs
using System;

static class Test
{
	private static T Greater<T>(T first, T second) where T : IComparable<T>
	{
		if(first.CompareTo(second) > 0)
			return first;

		return second;
	}

	public static void Main(string[] args)
	{
		string gs = Greater("Monday", "Tuesday");
		Console.WriteLine("Greater string = {0}", gs);

		double gd = Greater(9.75, 8.25);
		Console.WriteLine("Greater double = {0}", gd);

		Interval gi = Greater(new Interval(3, 45), new Interval(2, 30));
		Console.WriteLine("Greater Interval = {0}", gi);

	}
}

File: Coding Sample/Java/Fundamentals/Generic/Basics/GenMethodTest2.java
class GenMethodTest2{

	private static<T extends Comparable<T>> T greater(T first, T second){
		if(first.compareTo(second) > 0)
			return first;
		return second;
	}
	
	public static void main(String[] args){
		double gd = greater(6.78, 5.67);
		System.out.printf("Greater double = %s%n", gd);
		String gs = greater("Monday", "Tuesday");
		System.out.printf("Greater String = %s%n", gs);
		Interval gi = greater(new Interval(3, 45), new Interval(4, 56));
		System.out.printf("Greater Interval = %s%n", gi);
	}
}

